src/library/math/matrix.hpp

```diff
@@ -253,7 +253,7 @@ Matrix<R, R, T> rotz(const double& theta);
 
 }  // namespace libra
 
-#include "matrix_ifs.hpp"  // inline function definisions.
-#include "matrix_tfs.hpp"  // template function definisions.
+#include "matrix_inline_functions.hpp"
+#include "matrix_template_functions.hpp"
```

Comment:
functions よりは impl （implementation）とかの方が意図に即しているかなと思います
```suggestion
#include "matrix_impl_inline.hpp"
#include "matrix_impl_template.hpp"
```

Reply:
略語を使いたく無いので、`impl`はダメという議論になりました。  
`matrix_implementation_inline`や`matrix_inline_implementation`が良いということでしょうか？

Reply:
`functions` は性質であって意図ではないかなと思ったというかんじですね．`impl` みたいな一般的な略語もダメとなったんでしたっけ（略語がよくないのは毎回ドキュメントなり他のコードなりを見に行って意味を確認しなければならず認知負荷が不要に高くなるからだと思っています）．

Reply:
何が一般的なのか難しいということがいつも議論になるところです。  
プログラミング界隈では`impl = implementation`が一般的だとしても、プログラミングに詳しくない人から見ると、implicit?みたいな感じになってしまうこともあります。  

その逆のパターンで、衛星界隈では`rw = reaction wheel`ですが、衛星に詳しく無い人からみると random walk?みたいな感じになってしまうと思っています。（S2Eはその二つが混ざっていてさらにカオスでしたが）

ということで、結局は何が一般的かは業界や人に依存してしまうので、GNSSなどひどく長いもの以外、特に今回のように1単語のものについては書き出すのが良いと思っています。

Reply:
一応、[ここ](https://github.com/ut-issl/s2e-core/pull/311#issuecomment-1423521075)の議論が関連するので、メンションしておきます。  
@suzuki-toshihir0

Reply:
ありがとうございます．私はいまのところ `impl` を `implementation` に書き下したほうがS2Eの想定ユーザー（主に大学関係で，SWに必ずしも明るいとは限らない人？）に対しては親切だと思っています．

Reply:
理想的には、プログラミングに詳しくない衛星業界の人にも、衛星に詳しくないソフトウェア業界の人にもどちらにも参入障壁を低くしたいと思っているというのもあります。

Reply:
そうですね．どちらにも参入障壁を低くしたいとなると，両者に対して認知不可を下げるために書き下すべき語の範囲は広くなるように思いますね．

Reply:
なるほどたしかに．そうすると `inline_functions` とかの方がいいかもですね．よく考えたらこれはセマンティクス上の意図があっての分割ではないですし．

Reply:
あと，そもそも `math/matrix` ディレクトリを切った方が自然かもですね

Reply:
ありがとうございます。とりあえず`inline_functions`で進めようと思います。  
`math/matrix`に分けると、`math/matrix/matrix.hpp`がちょっと冗長な気はしています。vectorやquaternionも同様になるので、違うかなと思っていますがいかがでしょう？

Reply:
@sksat こちら、この部分が解決すればマージできると思っているのですが、ご意見いただけないでしょうか。私は、`math/matrix`のままで良いと思っています。

Reply:
これに関しては，同じ目的なもののファイルだけ分割したいようなものをファイルの prefix で管理することに個人的に少し違和感があったというだけなので，このままでもよいと思います．approve しますね．

Reply:
ありがとうございます！